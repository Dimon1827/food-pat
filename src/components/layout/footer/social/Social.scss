@import '@/assets/styles/variables';


.social {
  margin: 0;
  padding: 0;

  &--wrapper {
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    gap: 12px 24px;
    max-width: 132px;
    list-style: none;
    justify-content: space-between;
    padding: 20px 0 15px;
    margin: 20px auto;

    @media (min-width: $tablet-width) {
      border: none;
      margin: 0;
      margin-right: auto;
    }

    @media (max-width: $mobile-width) {
      &::before,
      &::after {
        content: "";
        position: absolute;
        bottom: 71%;
        left: 0;
        width: 100%;
        background-color: $darker-white-color;
        height: 1px;
      }

      &::after {
        bottom: 32%;
      }
    }
  }
}

.social__link {
  display: block;
  
  text-decoration: none;
  cursor: pointer;
  
  svg {
    fill: $primary-color;
    width: 22px;
    height: 28px;
  }

  &:hover svg {
    fill: $bg-basic-green;
  }

  &:active svg {
    fill: $bg-basic-green;
    opacity: 0.3;
  }
}
