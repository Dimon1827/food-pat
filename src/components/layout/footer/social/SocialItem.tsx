import { SocialItemProps } from './social.types';

function SocialItem({ socialData }: SocialItemProps) {
  return (
    <>
      {socialData.map((social, index) => {
        return (
          <li key={index} className="social__item" data-testid="social-item">
            <a href={social.href} className="social__link">
              {<social.component />}
              <span className="visually-hidden">{social.name}.</span>
            </a>
          </li>
        );
      })}
    </>
  );
}

export default SocialItem;
