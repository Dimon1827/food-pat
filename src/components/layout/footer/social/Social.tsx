import SocialItem from './SocialItem';
import './Social.scss';
import socialData from './socialData';

function Social() {
  return (
    <ul className="social social--wrapper" data-testid="social">
      <SocialItem socialData={socialData} />
    </ul>
  );
}

export default Social;
