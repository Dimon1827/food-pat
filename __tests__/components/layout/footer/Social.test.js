import { SlSocialVkontakte, SlSocialYoutube } from 'react-icons/sl';
import { screen, render } from '@testing-library/react';
import Social from '@/components/layout/footer/social/Social';
import SocialItem from '@/components/layout/footer/social/SocialItem';

describe('Social', () => {
  it('social renders correctly', () => {
    render(<Social />);
    expect(screen.getByTestId('social')).toMatchSnapshot();
  });
  it('socialItem renders correctly with 2 elements', () => {
    render(
      <SocialItem
        socialData={[
          {
            name: 'ВКонтакте',
            href: 'https://vk.com/id361320319',
            component: SlSocialVkontakte,
          },
          {
            name: 'Youtube',
            href: 'https://www.youtube.com/watch?v=QykPk380FkU',
            component: SlSocialYoutube,
          },
        ]}
      />
    );
    expect(screen.getAllByTestId('social-item')).toHaveLength(2);
  });
  it('socialItem renders correctly with 0 elements', () => {
    render(<SocialItem socialData={[]} />);
    const socialItems = screen.queryAllByTestId('social-item');
    expect(socialItems).toHaveLength(0);
  });
});
