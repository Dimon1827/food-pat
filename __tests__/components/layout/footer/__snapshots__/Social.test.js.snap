// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social social renders correctly 1`] = `
<ul
  class="social social--wrapper"
  data-testid="social"
>
  <li
    class="social__item"
    data-testid="social-item"
  >
    <a
      class="social__link"
      href="https://vk.com/id361320319"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3 248c0-38 26-53 58-55l149 1c9 0 17 6 20 15 34 110 76 178 126 255 3 6 8 9 13 9 4 0 8-2 11-7l3-11 1-173c0-25-12-29-40-33-11-2-18-12-18-22 0-2 0-4 1-6 14-43 58-65 120-65l56-1c46 0 88 20 88 79v227c4 3 8 5 13 5 8 0 18-5 26-18 52-73 111-160 119-206 0-2 1-3 2-5 11-22 39-37 51-41 2-1 5-2 9-2h155l10 1c15 0 26 10 31 19 9 14 7 29 8 35v7c-15 91-119 193-163 259-6 8-9 15-9 22 0 6 3 12 8 18l146 184c8 11 12 24 12 35 0 33-31 52-61 55l-17 1H779c-3 0-5 1-8 1-17 0-31-9-41-19-32-39-63-79-94-118-6-8-8-9-14-13-7 29-13 59-20 89l-3 17c-5 18-18 37-42 42l-14 1h-98C272 830 117 584 8 277c-3-8-5-19-5-29zm601 259c-26 0-55-15-55-43V234c0-27-12-37-45-37l-57 2c-32 0-50 5-65 15 23 11 44 26 44 68v176c-3 35-32 58-60 58-19 0-36-11-46-29-45-68-83-132-116-224l-9-26-133-1c-18 0-16 1-16 10 0 6 1 14 2 19l21 56c109 282 246 467 376 467h100c14 0 13-17 16-27l19-88c4-9 7-17 14-24 8-8 17-11 26-11 19 0 37 15 49 29l85 108c7 11 13 13 17 13h165c16 0 30-5 30-15 0-3-1-7-3-10L818 582c-12-15-17-30-17-45 0-16 6-32 16-46 42-63 132-153 153-227l3-13c-1-5-1-9-2-14H814c-10 4-18 10-24 18l-6 19c-23 64-86 152-131 213-15 14-32 20-49 20z"
        />
      </svg>
      <span
        class="visually-hidden"
      >
        ВКонтакте
        .
      </span>
    </a>
  </li>
  <li
    class="social__item"
    data-testid="social-item"
  >
    <a
      class="social__link"
      href="https://www.youtube.com/watch?v=QykPk380FkU"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M940.736 189.696C912.992 169.728 835.68 143.2 511.728 143.2c-347.152 0-398.656 30.464-415.185 40.432C8.575 236.479.223 470.447-.544 517.887.512 580.545 10.64 789.008 96.48 840.577c16.497 9.935 67.712 40.223 415.248 40.223 324.16 0 401.376-26.4 429.008-46.288 74.976-53.935 83.6-239.68 83.808-317.439-.192-62.528-6.752-271.872-83.807-327.376zm-37.376 592.88c-11.152 8.032-75.186 34.223-391.634 34.223-305.936 0-370.128-23.744-382.256-31.056-30.88-18.528-63.472-116.88-66.031-268.032 2.528-150.816 35.568-260.912 66.097-279.216 12.16-7.344 76.591-31.28 382.19-31.28 316.192 0 380.4 26.369 391.633 34.433 27.409 19.744 56.752 123.68 57.184 275.632-.432 154.336-29.968 245.712-57.184 265.296zM720.415 486.83L432.481 310.141a32.003 32.003 0 0 0-32.257-.464 32.016 32.016 0 0 0-16.288 27.872v353.44a32.016 32.016 0 0 0 32 32 32.02 32.02 0 0 0 16.527-4.592L720.4 541.645c9.6-5.807 15.472-16.19 15.472-27.407s-5.856-21.632-15.456-27.408zM447.953 634.301V394.204l194 120.032z"
        />
      </svg>
      <span
        class="visually-hidden"
      >
        Youtube
        .
      </span>
    </a>
  </li>
  <li
    class="social__item"
    data-testid="social-item"
  >
    <a
      class="social__link"
      href="https://www.youtube.com/watch?v=QykPk380FkU"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M228.88,26.19a9,9,0,0,0-9.16-1.57L17.06,103.93a14.22,14.22,0,0,0,2.43,27.21L72,141.45V200a15.92,15.92,0,0,0,10,14.83,15.91,15.91,0,0,0,17.51-3.73l25.32-26.26L165,220a15.88,15.88,0,0,0,10.51,4,16.3,16.3,0,0,0,5-.79,15.85,15.85,0,0,0,10.67-11.63L231.77,35A9,9,0,0,0,228.88,26.19Zm-61.14,36L78.15,126.35l-49.6-9.73ZM88,200V152.52l24.79,21.74Zm87.53,8L92.85,135.5l119-85.29Z"
        />
      </svg>
      <span
        class="visually-hidden"
      >
        Telegram
        .
      </span>
    </a>
  </li>
</ul>
`;
